source/Halibut/Transport/AuthorizedTcpConnectionsLimiter.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright 2012-2013 Octopus Deploy Pty. Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Concurrent;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using Halibut.Diagnostics;
+using Halibut.Exceptions;
+
+namespace Halibut.Transport
+{
+    public interface IAuthorizedTcpConnectionsLimiter
+    {
+        IDisposable ClaimAuthorizedTcpConnection(string thumbprint);
+    }
+
+    public class AuthorizedTcpConnectionsLimiter : IAuthorizedTcpConnectionsLimiter
+    {
+        readonly HalibutTimeoutsAndLimits timeoutsAndLimits;
+
+        //we use a StrongBox here so we can use Interlock to increment/decrement the value
+        ConcurrentDictionary<string, StrongBox<int>> authorizedConnectionCountPerThumbprint = new();
```

Comment:
What you need is [AtomicInteger](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/atomic/AtomicInteger.html)

Reply:
It's obvious @APErebus, just rewrite the whole thing in Java :)

Reply:
:shipit: